src/overlays/actors/ovl_En_Horse/z_en_horse.h

```diff
@@ -4,50 +4,192 @@
+#define ENHORSE_INGO_WON                   (1 << 23)  /*   0x80 0000 */
+#define ENHORSE_FLAG_24                    (1 << 24)  /*  0x100 0000 */
+#define ENHORSE_FLAG_25                    (1 << 25)  /*  0x200 0000 */
+#define ENHORSE_FLAG_26                    (1 << 26)  /*  0x400 0000 */
+#define ENHORSE_DRAW                       (1 << 27)  /*  0x800 0000 */
+#define ENHORSE_FLAG_28                    (1 << 28)  /* 0x1000 0000 */
+#define ENHORSE_FLAG_29                    (1 << 29)  /* 0x2000 0000 */
+#define ENHORSE_FLAG_30                    (1 << 30)  /* 0x4000 0000 */
+#define ENHORSE_FLAG_31                    (1 << 31)  /* 0x8000 0000 */
+
 struct EnHorse;
 
+typedef enum {
+    /* 0 */ PLAYER_DIR_FRONT_R,
+    /* 1 */ PLAYER_DIR_FRONT_L,
+    /* 2 */ PLAYER_DIR_BACK_R,
+    /* 3 */ PLAYER_DIR_BACK_L,
+    /* 4 */ PLAYER_DIR_SIDE_R,
+    /* 5 */ PLAYER_DIR_SIDE_L,
+} EnHorsePlayerDir;
+
+typedef enum {
+    /* 0 */ ENHORSE_ANIM_IDLE,
+    /* 1 */ ENHORSE_ANIM_WHINNEY,
+    /* 2 */ ENHORSE_ANIM_STOPPING,
+    /* 3 */ ENHORSE_ANIM_REARING,
+    /* 4 */ ENHORSE_ANIM_WALK,
+    /* 5 */ ENHORSE_ANIM_TROT,
+    /* 6 */ ENHORSE_ANIM_GALLOP,
+    /* 7 */ ENHORSE_ANIM_LOW_JUMP,
+    /* 8 */ ENHORSE_ANIM_HIGH_JUMP,
+} EnHorseAnimationIndex;
+
+typedef enum {
+    /* 0 */ HORSE_EPONA,
+    /* 1 */ HORSE_HNI,
+} HorseType;
+
+typedef void (*EnHorsePostdrawFunc)(struct EnHorse*, GlobalContext*);
+
 typedef struct EnHorse {
-    /* 0x000 */ Actor actor;
-    /* 0x014C */ s32 unk_14C;
-    /* 0x0150 */ char unk_150[0x008];
-    /* 0x0150 */ s32 unk_158;
-    /* 0x0150 */ char unk_15C[0x094];
-    /* 0x01F0 */ u32 unk_1F0;
-    /* 0x01F4 */ char unk_1F4[0x01C];
-    /* 0x0210 */ s32 unk_210;
-    /* 0x0214 */ f32 unk_214;
-    /* 0x0218 */ char unk_218[0x040];
-    /* 0x0258 */ Vec3f unk_258;
-    /* 0x0264 */ char unk_264[0x124];
-    /* 0x0388 */ s32 unk_388;
-    /* 0x038C */ char unk_38C[0x14];
-    /* 0x03A0 */ s32 unk_3A0;
-    /* 0x03A4 */ char unk_3A4[0x58];
+    /* 0x0000 */ Actor actor;
+    /* 0x014C */ EnHorseAction action;
+    /* 0x0150 */ s32 noInputTimer;
+    /* 0x0154 */ s32 noInputTimerMax;
+    /* 0x0158 */ s32 type;
+    /* 0x015C */ s8 bankIndex;
+    /* 0x0160 */ PSkinAwb skin;
+    /* 0x01F0 */ u32 flags;
+    /* 0x01F4 */ Vec3f lastPos;
+    /* 0x0200 */ s16 lastYaw;
+    /* 0x0204 */ s32 curRaceWaypoint;
+    /* 0x0208 */ s32 boostSpeed;
+    /* 0x020C */ s32 playerControlled;
+    /* 0x0210 */ s32 animationIdx;
+    /* 0x0214 */ f32 curFrame;
+    /* 0x0218 */ s32 soundTimer;
+    /* 0x021C */ Vec3f unk_21C;
+    /* 0x0228 */ Vec3f unk_228;
+    /* 0x0234 */ s32 unk_234;
+    /* 0x0238 */ u8 numBoosts;
+    /* 0x023C */ s32 boostRegenTime;
+    /* 0x0240 */ s32 boostTimer;
+    /* 0x0244 */ EnHorsePostdrawFunc postDrawFunc;
+
+    // The y coordinate of the floor under the front and back feet
+    /* 0x0248 */ f32 yFront;
+    /* 0x024C */ f32 yBack;
+
+    /* 0x0250 */ s16 followTimer;
```

Comment:
this is just a personal preference thing, but I like for there to be no line breaks in the struct just so its all one continuous block. would personally move the comment beside yFront and remove the breaks. not a requirement tho, just mentioning